### DISCUSSION TITLE
The energy and force values won't update

### DISCUSSION THREAD

**Discussion by mozhde** [MarsalekGroup/aml#4] [Q&A] [answered]
Hello,
I was playing around with the example to run a Qbc for a system of mine. My energy and force files are in CP2K format, and when I run the run-Qbc.py it generates the RuNNer format file properly. However, when I look at the "eval_train" and "predict" folders, I can see that the values of energy and forces in the "input.data" file are zero. Would you please help me find out what could go wrong?

Thank you

**Comment by cschran** [comment, marked as answer]
Dear mozhde,

this behavior is actually expected as the `eval_train` and `predict` folders are essentially intermediate directories within the QbC process of AML. The actual reference energies are saved internally for every structure and could be accessed within python, e.g. by a loop like this:
```
for s in structures:
    print(s.properties['reference'].energy)
```
Similarly, there will also be properties for the various individual NNP models under the key `predict-00X`.

If you want to get the force error of your model, you can make use of the functionality in `scores/forces.py` or have a look at the `score-all.py` script under `from-trajectroy-to-model/05-validation`.

**Comment by mozhde** [comment]
Dear Christoph

Thanks for your reply; it was constructive.
You are right; we don't need the reference values of the forces and
energies to evaluate the prediction of the committee members with respect
to each other.
If it is fine with you, I have another question to ask you; Suppose I want
to get the score given by function " evaluate_cda_forces" for an already
trained committee. I can see you did it in class "Qbc," but you've defined
a new blank committee and don't use an already trained one. More precisely,
I want to query a committee outside the "Qbc" iteration to pick the highest
disagreement for my data manually. Would it be possible?

Regards

**Comment by cschran** [comment]
If you have obtained the prediction of the structures, you can use the function `select_highest_error`, e.g. as done in the QbC procedure:
```
idx_selected = structures_candidate.select_highest_error(
    n=n_add, label_prop='pred*', f_reduce=moment_std, name_prop='forces')
```

This also allows you to change the function `f_reduce` that reduces the force disagreement into a single score per structure.
